### PR TITLE
Retry access token refresh errors

### PR DESCRIPTION
**Pull request by krames**
We are still seeing HTTP 401 exceptions containing the body `Access token expired. Use your refresh token to obtain a new access token.`

This PR detects these requests and then attempts to retry them 3 times using an incrmental back of of (count * 0.1 seconds). After the 3rd attempt is reached we just throw a `BitBucket::Error::RefreshToken` exception.

### CODE REVIEW

File: lib/bitbucket_rest_api/repos/webhooks.rb [hunk 1]
```diff
@@ -62,7 +62,7 @@ def get(user_name_or_project_key, repo_name, hook_uuid)
       _validate_user_repo_params(user, repo) unless user? && repo?
 
       get_request(
-        "/2.0/repositories/#{user_name}/#{repo_name}/hooks/#{hook_uuid}"
+        "/2.0/repositories/#{user_name_or_project_key}/#{repo_name}/hooks/#{hook_uuid}"
```
Comment (krames) [inline lib/bitbucket_rest_api/repos/webhooks.rb]: Fixes existing failing test.

File: lib/bitbucket_rest_api/request.rb [hunk 2]
```diff
@@ -28,6 +28,20 @@ def delete_request(path, params={}, options={})
       request(:delete, path, params, options)
     end
 
+    def retry_token_refresh_errors
+      count = 0
+      begin
+        yield
+      rescue BitBucket::Error::RefreshToken
+        count += 1
+        if count < 3
```
Comment (markphelps) [inline lib/bitbucket_rest_api/request.rb]: I think you want `<= 3` here if you want it retry 3 times after the first failed attempt.

File: lib/bitbucket_rest_api/response/raise_error.rb [hunk 3]
```diff
@@ -11,6 +13,9 @@ def on_complete(env)
       when 400
         raise BitBucket::Error::BadRequest.new(env)
       when 401
+        if env[:body] =~ REFRESH_TOKEN
```
Comment (markphelps) [inline lib/bitbucket_rest_api/response/raise_error.rb]: I guess there's no way to determine if its a 'retriable' 401 without using regex to parse the message?

**Review by markphelps — APPROVED** [review]

File: lib/bitbucket_rest_api/response/raise_error.rb (hunk 3 above)
Comment (krames) [reply lib/bitbucket_rest_api/response/raise_error.rb]: Not that I can think of and unfortunately, I have not been able to reproduce this locally so I am sort of flying blind.

**Pull request merged**